caab-service/src/main/java/uk/gov/laa/ccms/caab/api/entity/converter/BooleanStringConverter.java

```diff
@@ -0,0 +1,115 @@
+package uk.gov.laa.ccms.caab.api.entity.converter;
+
+import jakarta.persistence.AttributeConverter;
+import java.util.Arrays;
+import java.util.stream.Stream;
+import org.hibernate.dialect.Dialect;
+import org.hibernate.type.descriptor.converter.spi.BasicValueConverter;
+import org.hibernate.type.descriptor.java.BooleanJavaType;
+import org.hibernate.type.descriptor.java.JavaType;
+import org.hibernate.type.descriptor.java.StringJavaType;
+import org.hibernate.type.descriptor.jdbc.JdbcType;
+import org.springframework.beans.factory.annotation.Value;
+
+/**
+ * Configurable AttributeConverter to handle conversion to and from Boolean and DB
+ * representation of Boolean values.
+ */
+public class BooleanStringConverter implements
+    AttributeConverter<Boolean, String>, BasicValueConverter<Boolean, String> {
+
+  public static final BooleanStringConverter INSTANCE = new BooleanStringConverter();
+
+  /**
+   * Array of values which should resolve to a Boolean.TRUE.
+   */
+  @Value("${laa.ccms.caab.converters.boolean.true_values}")
+  protected String[] trueValues;
+
+  /**
+   * Array of values which should resolve to a Boolean.FALSE.
+   */
+  @Value("${laa.ccms.caab.converters.boolean.false_values}")
+  protected String[] falseValues;
+
+  public BooleanStringConverter() {
+    super();
+  }
+
+  /**
+   * Constructor which takes an array of true and false values.
+   * Added for testing purposes only.
+   *
+   * @param trueValues - the true values.
+   * @param falseValues - the false values.
+   */
+  protected BooleanStringConverter(final String[] trueValues, final String[] falseValues) {
+    this.trueValues = trueValues;
+    this.falseValues = falseValues;
+  }
+
+  /**
+   * Get all of the values supported by this converter.
+   *
+   * @return a concatenated list of true and false values.
+   */
+  protected String[] getValues() {
+    return Stream.concat(
+        Arrays.stream(trueValues),
+        Arrays.stream(falseValues)).toArray(String[]::new);
+  }
+
+  /**
+   * Convert the relational form to the domain value.
+   *
+   * @param relationalForm - the relational form.
+   *
+   * @return Boolean representation of relational form, or null.
+   */
+  public Boolean toDomainValue(String relationalForm) {
+    if (relationalForm == null) {
+      return null;
+    }
+
+    return Arrays.asList(trueValues).contains(relationalForm)
+        ? Boolean.TRUE :
+        (Arrays.asList(falseValues).contains(relationalForm) ? Boolean.FALSE : null);
```

Comment:
It may be worth raising an exception and/or logging if there is a non-null value that is not in the list.

Reply:
Agreed - reworked this method and the associated tests